lib/note-detail.jsx

```diff
@@ -19,36 +19,61 @@ export default React.createClass( {
 
 	componentWillMount: function() {
 		this.queueNoteSave = debounce( this.saveNote, saveDelay );
-		this.noteEditor = uninitializedNoteEditor;
+		this.focus = () => this.refs.editor.focus();
+		this.state = {
+			editorState: EditorState.createEmpty( urlDecorator )
+		};
 	},
 
 	componentDidMount: function() {
 		// Ensures note gets saved if user abruptly quits the app
 		window.addEventListener( 'beforeunload', this.queueNoteSave.flush );
-	},
-
-	initializeNoteEditor: function( noteEditor ) {
-		this.noteEditor = noteEditor;
+		this.readNoteFromProps( this.props );
 	},
 
 	isValidNote: function( note ) {
 		return note && note.id;
 	},
 
-	componentWillReceiveProps: function() {
+	hasContentChanged( editorState, note ) {
+		return get( note, 'data.content', '' )
+			!== editorState.getCurrentContent().getPlainText( '\n' )
+	},
+
+	handleEditorStateChange( editorState ) {
+		this.setState( { editorState } );
+		// only queue note save if it was the content that changed
+		if ( this.hasContentChanged( editorState, this.props.note ) ) {
+			this.queueNoteSave();
+		}
```

Comment:
1) form looking at the draft.js docs it appears that a `ContentState` contains info about both the contents and the current `SelectionState` of the editor. To avoid triggering a save when all the user changed was the selection, I think we need to compare the actual content which for our purposes (since that's how it is saved) is what's returned from `getPlainText`. Now the question is, do we want to compare against the note in the props, or against the previous `ContentState`?

2) great point about doing this in a setState callback, will change